signature/jws/jwt.go

```diff
@@ -0,0 +1,152 @@
+package jws
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/notaryproject/notation-core-go/signature"
+)
+
+// signingMethod is the interface for jwt.SigingMethod with additional method to
+// access certificate chain after calling Sign().
+type signingMethod interface {
```

Comment:
Where are we using `signingMethod`?

Reply:
In order to use JWT Sign, we need to provide a jwt.SigningMethod. `remoteSigningMethod` and `localSigningMethod` implement `signingMethod` interface. This interface is designed for wrapping the Signer to be a jwt.SigningMethod, also mitigating the difference between `local signer` and `remote signer`. For example, the `certificate chain` and `private key` are included in signature.LocalSigner, but for remote signer, the are provided by external plugin and can only be accessed after signing, so by implementing CertificateChain() and PrivateKey() can mitigate the difference between them and reduce the `if...else...` check.